src/web/app/components/copy-course-modal/copy-course-modal.component.ts

```diff
@@ -0,0 +1,104 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+import { StatusMessageService } from '../../../services/status-message.service';
+import { TimezoneService } from '../../../services/timezone.service';
+import { FeedbackSession } from '../../../types/api-output';
+import { COURSE_ID_MAX_LENGTH } from '../../../types/field-validator';
+
+interface Timezone {
+  id: string;
+  offset: string;
+}
+
+const formatTwoDigits: Function = (n: number): string => {
+  if (n < 10) {
+    return `0${n}`;
+  }
+  return String(n);
+};
+
+/**
+ * Copy current course modal.
+ */
+@Component({
+  selector: 'tm-copy-course-modal',
+  templateUrl: './copy-course-modal.component.html',
+  styleUrls: ['./copy-course-modal.component.scss'],
+})
+export class CopyCourseModalComponent implements OnInit {
+
+  // const
+  COURSE_ID_MAX_LENGTH: number = COURSE_ID_MAX_LENGTH;
+
+  @Input()
+  sessionsInCourse: FeedbackSession[] = [];
+
+  timezones: Timezone[] = [];
```

Comment:
Its probably better to group these attributes into a type e.g CopyCourseModalAttributes